### ISSUE TITLE
TransportSerializationException during rapse IndicesStatsResponse

### ISSUE THREAD

**Issue by hleb-albau** [strapdata/elassandra#147]
Hi,

elassandra 5.5.0.8, driver 5.5.3
next code produce TransportSerializationException

```
val request = IndicesStatsRequest().all()
val statsResponse = elasticClient.admin().indices().stats(request).actionGet()
```

```
TransportSerializationException[Failed to deserialize response of type [org.elasticsearch.action.admin.indices.stats.IndicesStatsResponse]]; nested: IOException[Invalid vInt ((ffffffff & 0x7f) << 28) | fffffff];
	at org.elasticsearch.transport.TcpTransport.handleResponse(TcpTransport.java:1431)
	at org.elasticsearch.transport.TcpTransport.messageReceived(TcpTransport.java:1403)
	at org.elasticsearch.transport.netty4.Netty4MessageChannelHandler.channelRead(Netty4MessageChannelHandler.java:74)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:310)
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:297)
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:413)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:265)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1334)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:926)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:134)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:644)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysPlain(NioEventLoop.java:544)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:498)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:458)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:858)
	at java.base/java.lang.Thread.run(Thread.java:844)
Caused by: java.io.IOException: Invalid vInt ((ffffffff & 0x7f) << 28) | fffffff
	at org.elasticsearch.common.io.stream.StreamInput.readVInt(StreamInput.java:206)
	at org.elasticsearch.index.engine.SegmentsStats.readFrom(SegmentsStats.java:362)
	at org.elasticsearch.common.io.stream.StreamInput.readOptionalStreamable(StreamInput.java:701)
	at org.elasticsearch.action.admin.indices.stats.CommonStats.<init>(CommonStats.java:243)
	at org.elasticsearch.action.admin.indices.stats.ShardStats.readFrom(ShardStats.java:92)
	at org.elasticsearch.action.admin.indices.stats.ShardStats.readShardStats(ShardStats.java:85)
	at org.elasticsearch.common.io.stream.StreamInput.readArray(StreamInput.java:685)
	at org.elasticsearch.action.admin.indices.stats.IndicesStatsResponse.readFrom(IndicesStatsResponse.java:140)
	at org.elasticsearch.transport.TcpTransport.handleResponse(TcpTransport.java:1428)
	... 23 more

```

**Comment by vroyer** [comment]
Yes, elassandra introduced a tokenRangesBitsetMemoryInBytes in stats that is not compatible with the standard elasticsearch client.

Using the elasticsearch client from elassandra should solve this issue, by installing elassandra in your local repository with gradle publishToMavenLocal -Dbuild.snapshot=false
Then use the following dependency :

	<dependency>
            <groupId>com.strapdata.elasticsearch</groupId>
            <artifactId>elasticsearch</artifactId>
            <version>5.5.0</version>
        </dependency>.

**Comment by hleb-albau** [comment]
As far, as i understand, you already publish[ this lib](https://mvnrepository.com/artifact/com.strapdata.elasticsearch/elasticsearch/5.5.0)?

**Comment by vroyer** [comment]
Yes, but the published version was the 5.5.0.1 and i can’t update-it because of the sonatype policy.
I would be able to update-it with a version number with 4 digits (ex: 5.5.0.x), but this won’t match any more the elasticsearch version number for plugin management.
This is the reason why a recommend to build your own up-to-date transport client library to solve your issue.

**Comment by hleb-albau** [comment]
Okey, thanks for clarifying. Also, its possible to have 4 digits version for maven artifacts. [For example, look at kafka artifacts](https://mvnrepository.com/artifact/org.apache.kafka/kafka).
I can investigate and make pr if you need.
